### PR TITLE
refactor(core): Improve `MapErr` for `transport`; improve docs and arrange code

### PR DESCRIPTION
**Pull request by yjhmelody**
## Description

- Format code and improve docs.
- Remove some unused fields.
- Rename some modules to keep the same style.
- Define `MapErrFuture` to keep the same code style.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by jxs — COMMENTED** [review]
Hi, and thanks for your interest! Overall looks good, can we maintain the function order for less diffs?

File: core/src/transport.rs [hunk 1]
```diff
@@ -35,11 +35,11 @@ use std::{
 };
 
 pub mod and_then;
-pub mod choice;
 pub mod dummy;
 pub mod map;
 pub mod map_err;
 pub mod memory;
+pub mod or;
```
Comment (thomaseizinger) [inline core/src/transport.rs:42]: This is a breaking change. I'd like to have this non-breaking please. Should be easy with another module and type aliases.

File: core/src/transport/and_then.rs [hunk 2]
```diff
@@ -78,7 +78,6 @@ where
                     role_override: Endpoint::Dialer,
                 },
             )),
-            _marker: PhantomPinned,
```
Comment (thomaseizinger) [inline core/src/transport/and_then.rs:81]: Is this safe? I am sure this wasn't added without reason.

@mxinden Do you know why we phantom pin this type?

File: core/src/transport/map_err.rs [hunk 3]
```diff
@@ -76,73 +77,49 @@ where
...
+    fn address_translation(&self, server: &Multiaddr, observed: &Multiaddr) -> Option<Multiaddr> {
+        self.transport.address_translation(server, observed)
     }
 }
 
-/// Dialing future for `MapErr`.
+/// Future for `MapErr`.
 #[pin_project::pin_project]
-pub struct MapErrDial<T: Transport, F> {
+pub struct MapErrFuture<T, Fut, F> {
```
Comment (thomaseizinger) [inline core/src/transport/map_err.rs:112]: This is also breaking. Please add a type alias to make it non-breaking with a deprecation attribute.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Thanks!

There are several unrelated things happening here. We squash merge PRs so I'd appreciate it if we can split those i.e. do a separate PR for reordering the functions etc

**Comment by thomaseizinger** [comment]
Closing due to inactivity.

**Pull request closed without merging**